pkg/oauth2google/auth_cntrl.go

```diff
@@ -79,21 +79,27 @@ func (c *AuthCntrl) Callback(ce echo.Context) (err error) {
 
 // Logout by invalidating cookies
 func (c *AuthCntrl) Logout(ce echo.Context) (err error) {
-	secureTokenCookie := &http.Cookie{Name: "secure_token", MaxAge: -1, Path: "/"}
-	ce.SetCookie(secureTokenCookie)
-
-	tokenCookie := &http.Cookie{Name: "token", MaxAge: -1, Path: "/"}
-	ce.SetCookie(tokenCookie)
-
+	ce.SetCookie(&http.Cookie{Name: "secure_token", MaxAge: -1, Path: "/"})
+	ce.SetCookie(&http.Cookie{Name: "token", MaxAge: -1, Path: "/"})
 	return ce.Redirect(http.StatusSeeOther, c.LogoutRedirect)
 }
 
 // Middleware for google social login
 func (c *AuthCntrl) Middleware() echo.MiddlewareFunc {
-	jwtCfg := middleware.DefaultJWTConfig
-	jwtCfg.SigningKey = []byte(c.JWTSecret)
-	jwtCfg.TokenLookup = "cookie:secure_token"
-	return middleware.JWTWithConfig(jwtCfg)
+	return func(next echo.HandlerFunc) echo.HandlerFunc {
+		return func(ce echo.Context) error {
+			cfg := middleware.DefaultJWTConfig
+			cfg.SigningKey = []byte(c.JWTSecret)
+			cfg.TokenLookup = "cookie:secure_token"
+
+			if err := middleware.JWTWithConfig(cfg)(next)(ce); err != nil {
+				log.Warnf("JWT Error: %s", err.Error())
+				return c.Logout(ce)
+			}
```

Comment:
Not so sure.
But if it works for now, it is okay as workaround.
My guess is setting `cfg.ErrorHandler` will be sufficient

Reply:
ErrorHandle shows the error on the response but not clear the cookie. This makes UI always thought that the session is logged-in yet the token actually expired. Maybe don't need to redirect but only clear cookie when JWT error.